File: e2e-vstest-selenium/LoginTest.cs
namespace e2e_vstest_selenium;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;

[TestClass]
public class LoginTest
{
    static IWebDriver driver = null;

    [ClassInitialize()]
    public static void BeforeAll(TestContext context) {
        driver = new ChromeDriver();
        driver.Navigate().GoToUrl("http://travel.agileway.net");
    }

    [TestInitialize]
    public void BeforeTest() {
        driver.Navigate().GoToUrl("http://travel.agileway.net/login");
    }


    [TestCleanup]
    public void AfterTest() {
    // run after each test case
    }

    [ClassCleanup()]
    public static void AfterAll() {
      driver.Quit();
    }

    [TestMethod]
    public void TestLoginOK()
    {
        driver.FindElement(By.Name("username")).SendKeys("agileway");
        driver.FindElement(By.Name("password")).SendKeys("testwise");
        driver.FindElement(By.Name("password")).Submit();
        Assert.IsTrue(driver.PageSource.Contains("Signed in!"));
    }

    [TestMethod]
    public void TestLoginFailed()
    {
        driver.FindElement(By.Name("username")).SendKeys("agileway");
        driver.FindElement(By.Name("password")).SendKeys("badpass");
        driver.FindElement(By.Name("password")).Submit();
        Assert.IsTrue(driver.PageSource.Contains("Invalid email or password"));
    }
}

File: e2e-junit-selenium/src/tests/LoginTest.java
package tests;

import org.junit.*;
import org.junit.Assert.*;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;

import pages.*;

public class LoginTest {
    static WebDriver driver;

    @BeforeClass
    public static void setUp() throws Exception {
        driver = new ChromeDriver();
        driver.get("https://travel.agileway.net");
    }

    @AfterClass
    public static void tearDown() {
        driver.quit();
    }

    @Test
    public void testSignInFailed() {
        LoginPage loginPage = new LoginPage(driver);
        loginPage.enterUsername("agileway");
        loginPage.enterPassword("guess");
        loginPage.clickSignIn();
        Assert.assertEquals("Invalid email or password", driver.findElement(By.id("flash_alert1")).getText() ); 
    }

    @Test
    public void testSignInOK() {
        LoginPage loginPage = new LoginPage(driver);
        loginPage.enterUsername("agileway");
        loginPage.enterPassword("testwise");
        loginPage.clickSignIn();
        Assert.assertTrue( driver.findElement(By.tagName("body")).getText().contains("Signed in!") );
    }


}